CSAR.lua

```diff
@@ -1597,13 +1646,91 @@ function csar.checkOnboard(_unitName)
     end
 end
 
+function csar.addweight( _heli )
+  local cargoWeight = 0
+  
+  local _heliName =  _heli:getName()
+  if ctld ~= nil and ctld.troopWeight ~= nil then
+      -- TODO Count CTLD troops
+          
+  end
+  ctld.troopWeight = 100
+  if csar.inTransitGroups[_heliName] then
+    local csarcount = 0
+    for _, _group in pairs(csar.inTransitGroups[_heliName]) do
+        csarcount = csarcount + 1
+    end
+    cargoWeight = cargoWeight + csar.weight * csarcount
+  end
+  
+  trigger.action.setUnitInternalCargo(_heli:getName(),0 ) -- Set To  to recalculate 
+  trigger.action.setUnitInternalCargo(_heli:getName(), cargoWeight)
+  
 
+end
 -- Adds menuitem to all medevac units that are active
 function csar.addMedevacMenuItem()
     -- Loop through all Medevac units
 
     timer.scheduleFunction(csar.addMedevacMenuItem, nil, timer.getTime() + 5)
 
+    local _allHeliGroups = coalition.getGroups(coalition.side.BLUE, Group.Category.HELICOPTER)
```

Comment:
People have done CSAR with non helicopters like the TF-51 :D

Reply:
That is for the prefix purpose..   Since I added back the Unit list  i don't think to loop all the planes here as well will be needed.. I added some clarifications that the prefix is meant for Helicopter units.